bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/flamingos2/F2ScienceAreaGeometry.scala

```diff
@@ -0,0 +1,109 @@
+package edu.gemini.spModel.gemini.flamingos2
+
+import java.awt.Shape
+import java.awt.geom.{Rectangle2D, Area, Ellipse2D}
+
+import edu.gemini.shared.util.immutable.DefaultImList
+import edu.gemini.spModel.gemini.flamingos2.Flamingos2.{FPUnit, LyotWheel}
+import edu.gemini.spModel.inst.ScienceAreaGeometry
+
+import scala.collection.JavaConverters._
+
+class F2ScienceAreaGeometry(inst0: Flamingos2) extends ScienceAreaGeometry[Flamingos2] {
+  import F2ScienceAreaGeometry._
+
+  override def geometry: List[Shape] = {
+    Option(inst0).toList.flatMap { inst =>
+      lazy val plateScale = inst.getLyotWheel.getPlateScale
+      lazy val scienceAreaWidth = {
+        val scienceAreaArray = inst.getScienceArea
+        scienceAreaArray(0)
+      }
+      inst.getFpu match {
+        case Flamingos2.FPUnit.FPU_NONE    => List(imagingFOV(plateScale))
+        case Flamingos2.FPUnit.CUSTOM_MASK => List(mosFOV(plateScale))
+        case _ if inst.getFpu.isLongslit   => List(longSlitFOV(plateScale, scienceAreaWidth))
+        case _                             => Nil
+      }
+    }
+  }
+
+  // We need to override this due to type system.
+  override def geometryAsJava: edu.gemini.shared.util.immutable.ImList[Shape] =
+    DefaultImList.create(geometry.asJava)
+
+  def scienceAreaDimensions: (Double, Double) = {
+    Option(inst0).map { inst =>
+      val lyotWheel = inst.getLyotWheel
+      if (Set(LyotWheel.OPEN, LyotWheel.HIGH, LyotWheel.LOW).contains(lyotWheel)) {
```

Comment:
Perhaps a match on `lyotWheel` instead of making a `Set` there.  No big deal either way.

````scala
import LyotWheel._
lyotWheel match {
  case OPEN | HIGH | LOW =>
  case _ =>
}
````